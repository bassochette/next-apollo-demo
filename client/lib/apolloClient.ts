import { ApolloClient, InMemoryCache } from "@apollo/client";
import getConfig from "next/config";

const { publicRuntimeConfig } = getConfig();

const apolloClient = new ApolloClient({
  uri: publicRuntimeConfig.GRAPHQL_URL || "http://localhost:5000/graphql",
  cache: new InMemoryCache(),
});

export const client = apolloClient;
export default apolloClient;
