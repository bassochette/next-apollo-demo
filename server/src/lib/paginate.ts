import { Contact, PaginatedContacts } from "../contact/contact.interface";

export const paginate = (
  data: Contact[],
  page = 0,
  pageSize = 20
): PaginatedContacts => {
  return {
    contacts: data.slice(page * pageSize, page * pageSize + pageSize),
    page,
    pageSize,
    pageCount: Math.ceil(data.length / pageSize),
  };
};
