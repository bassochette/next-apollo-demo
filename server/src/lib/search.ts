import { filter } from "lodash";
import { Contact } from "../contact/contact.interface";

export const search = (contacts: Contact[], query: string): Contact[] => {
  return filter(contacts, (contact) => {
    return contact.name.toLowerCase().includes(query.toLowerCase());
  });
};
