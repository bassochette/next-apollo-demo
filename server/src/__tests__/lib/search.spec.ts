import { contactsData } from "../contacts.data";
import { search } from "../../lib/search";

describe("lib/search", () => {
  it("match snapshot", () => {
    const result = search(contactsData, "ma");
    expect(JSON.stringify(result)).toMatchSnapshot();
  });
});
