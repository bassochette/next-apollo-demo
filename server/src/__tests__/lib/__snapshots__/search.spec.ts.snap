// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/search match snapshot 1`] = `"[{\\"__typename\\":\\"Contact\\",\\"address\\":\\"47 Duane Roads Suite 323\\\\nPort Jedidiahview, TX 00121-1578\\\\n86367 South Houstonton\\\\nParaguay\\",\\"email\\":\\"Flavio.Stroman@Farrell.name\\",\\"name\\":\\"Monserrat Mayert\\",\\"phone\\":\\"833-621-4053\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"15 Mckenna Ridges Suite 920\\\\nNew Lavinia, KY 35757-5150\\\\n29575-2811 Cadenburgh\\\\nKorea\\",\\"email\\":\\"Kuhn_Ernie@Nathanial.biz\\",\\"name\\":\\"Maye Carter\\",\\"phone\\":\\"298-239-2516\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"069 Oberbrunner Fort Suite 363\\\\nWest Leola, IA 98568\\\\n62777 Bednarstad\\\\nPapua New Guinea\\",\\"email\\":\\"Milford.Stehr@Barton.io\\",\\"name\\":\\"Demarco Glover\\",\\"phone\\":\\"214-310-7601\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"71 Satterfield Junctions\\\\nDiegoberg, VT 93650\\\\n09166 East Kasandra\\\\nAnguilla\\",\\"email\\":\\"Roma.Reichert@Douglas.com\\",\\"name\\":\\"Manley Rodriguez\\",\\"phone\\":\\"301-570-0138\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"1174 Ruben Street Suite 181\\\\nWest Samson, SC 04244-2236\\\\n46794 West Ted\\\\nNauru\\",\\"email\\":\\"Laila_Ziemann@Smith.name\\",\\"name\\":\\"Marshall Green\\",\\"phone\\":\\"887-333-1856\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"0151 Verdie Cove Suite 585\\\\nStantonberg, KS 17930\\\\n96428-6327 East Priscillahaven\\\\nKuwait\\",\\"email\\":\\"Evan.Abshire@Alysha.info\\",\\"name\\":\\"Mabel Okuneva\\",\\"phone\\":\\"199-625-5127\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"5651 Legros Glens\\\\nMillsville, MA 96700\\\\n45799-5027 Deonteport\\\\nNorway\\",\\"email\\":\\"Funk.Mariane@Marcus.io\\",\\"name\\":\\"Maia Anderson\\",\\"phone\\":\\"126-631-4167\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"9944 Davion Plains\\\\nSelinaton, DC 49770\\\\n33895-0052 North Cody\\\\nEgypt\\",\\"email\\":\\"Kub_Samson@Forest.us\\",\\"name\\":\\"Oda Stiedemann\\",\\"phone\\":\\"609-769-3701\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"59 Runolfsdottir Station\\\\nBreitenbergberg, MD 66297\\\\n13301 Magdalenfort\\\\nSaint Vincent and the Grenadines\\",\\"email\\":\\"Cora_Pfannerstill@Rosanna.com\\",\\"name\\":\\"Maxie Toy\\",\\"phone\\":\\"980-697-4345\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"3307 Micheal Highway Apt. 479\\\\nPredovicmouth, WI 31698-8840\\\\n56823 Merlhaven\\\\nLesotho\\",\\"email\\":\\"Volkman_Dominic@Christiansen.biz\\",\\"name\\":\\"Lori Mayert\\",\\"phone\\":\\"590-675-2123\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"883 Collins Mountain\\\\nNorth Jackyton, AK 82937-8832\\\\n19863-2940 East Joannychester\\\\nKorea\\",\\"email\\":\\"Skye.Weissnat@King.org\\",\\"name\\":\\"Margaret Graham\\",\\"phone\\":\\"026-770-8130\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"823 Curtis Fort Apt. 619\\\\nLake Johnpaul, MN 97700\\\\n01180 South Owen\\\\nLiberia\\",\\"email\\":\\"Berneice.Flatley@Robel.info\\",\\"name\\":\\"Amelie Maggio\\",\\"phone\\":\\"923-395-4397\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"96 Vida Villages\\\\nMaiamouth, WV 21809\\\\n69837-8273 Port Jovany\\\\nCentral African Republic\\",\\"email\\":\\"Paucek.Ashleigh@Juliet.biz\\",\\"name\\":\\"Emmanuelle Becker\\",\\"phone\\":\\"637-367-4189\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"620 Delmer Summit\\\\nImamouth, PA 46981\\\\n78787-2808 McKenziechester\\\\nSvalbard & Jan Mayen Islands\\",\\"email\\":\\"Dorothea.Morar@yahoo.com\\",\\"name\\":\\"Manley Kautzer\\",\\"phone\\":\\"315-483-5388\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"0824 Bergstrom Junction\\\\nSouth Rafaelafurt, MN 77917\\\\n93865-9855 Jarrellchester\\\\nMacao\\",\\"email\\":\\"Jamison_Kiehn@yahoo.com\\",\\"name\\":\\"Tomas Spencer\\",\\"phone\\":\\"577-087-3732\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"790 Owen Mountains Suite 124\\\\nAdolphushaven, WA 71881\\\\n57235 Venaville\\\\nGuyana\\",\\"email\\":\\"Osinski.Modesta@yahoo.com\\",\\"name\\":\\"Alejandra Eichmann\\",\\"phone\\":\\"041-448-3672\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"29 Elmira Views Apt. 958\\\\nRebaton, IA 57721\\\\n71823-8525 Britneyberg\\\\nNorfolk Island\\",\\"email\\":\\"Prohaska.Antonietta@yahoo.com\\",\\"name\\":\\"Declan Weimann\\",\\"phone\\":\\"106-310-9742\\"},{\\"__typename\\":\\"Contact\\",\\"address\\":\\"11 Noah Spring\\\\nNorth Darrick, CA 37731\\\\n99074 West Jorgemouth\\\\nIsle of Man\\",\\"email\\":\\"Merl.Sauer@gmail.com\\",\\"name\\":\\"Tristin Stroman\\",\\"phone\\":\\"787-436-9389\\"}]"`;
